src/Build.OM.UnitTests/Construction/SolutionFile_Tests.cs

```diff
@@ -16,6 +16,84 @@ namespace Microsoft.Build.UnitTests.Construction
     /// </summary>
     public class SolutionFile_Tests
     {
+        [Theory]
+        [InlineData(@"
```

Comment:
It's not obvious to me what's wrong with these filters. Can you either

1. abstract the current method to a helper and make each of these a named method describing the problem
2. specify additional data in the theory input to give names (or inner exception types?)

That would also allow you to have more detailed checks on the assertion--"doesn't accept bad input of type X" is bare minimum and I'd like to test that the correct, specific, diagnosable errors are produced.